Getting exit code 0 regardless of the test results

Hi,

When running tests with v2.0.0, i'm always getting exit code 0, regardless of if the tests have failed or not.
I've verified that in v1.5.4 when the tests failed, the exit code was 2.

Here you go
**Test**
```
var should = require('should');

it('test mochawesome', function () {
   true.should.equal(false) 
});
```

**With mochawesome 2.0.0**
```
bash-3.2$ node_modules/mocha/bin/mocha test/mytest.js --reporter mochawesome; echo $?
  1) test mochawesome

  0 passing (14ms)
  1 failing

  1)  test mochawesome:

      AssertionError: expected true to be false
      + expected - actual

      -true
      +false

      at Assertion.fail (node_modules/should/cjs/should.js:231:17)
      at Assertion.value (node_modules/should/cjs/should.js:305:19)
      at Context.<anonymous> (test/mytest.js:4:16)
[mochawesome] Report JSON saved to mochawesome-reports/mochawesome.json

[mochawesome] Report HTML saved to mochawesome-reports/mochawesome.html
0
bash-3.2$
```

**Without mochawesome**
```
bash-3.2$ node_modules/mocha/bin/mocha test/mytest.js ; echo $?
  1) test mochawesome

  0 passing (14ms)
  1 failing

  1)  test mochawesome:

      AssertionError: expected true to be false
      + expected - actual

      -true
      +false

      at Assertion.fail (node_modules/should/cjs/should.js:231:17)
      at Assertion.value (node_modules/should/cjs/should.js:305:19)
      at Context.<anonymous> (test/mytest.js:4:16)
1
bash-3.2$
```

--- a/src/mochawesome.js
+++ b/src/mochawesome.js
@@ -289,7 +289,7 @@ function saveFile(filename, data) {
  * @param {Function} exit
  */
 
-async function done(output, config, exit) {
+async function done(output, config, failures, exit) {
   const { reportJsonFile, reportHtmlFile } = config;
   try {
     // Save the JSON to disk
@@ -300,10 +300,10 @@ async function done(output, config, exit) {
     await marge.create(output, config);
     log(`Report HTML saved to ${reportHtmlFile}`, null, config);
 
-    exit();
+    exit(failures);
   } catch (err) {
     log(err, 'error', config);
-    exit();
+    exit(failures);
   }
 }
 
@@ -317,7 +317,7 @@ async function done(output, config, exit) {
 function Mochawesome(runner, options) {
   // Done function will be called before mocha exits
   // This is where we will save JSON and generate the report
-  this.done = (failures, exit) => done(this.output, this.config, exit);
+  this.done = (failures, exit) => done(this.output, this.config, failures, exit);
 
   // Reset total tests counter
   totalTestsRegistered = 0;

--- a/test-functional/mocha.opts
+++ b/test-functional/mocha.opts
@@ -1,5 +1,4 @@
 --stack-trace
 --compilers js:babel-core/register
---require babel-polyfill
 --require should
 --reporter mochawesome

